### PR TITLE
chore: add struct tags and update docs

### PR DESCRIPTION
**Pull request by Chief-Rishab**
Updating README for following plugins: 
- [x] MySQL (`identifier` field isn't required for URN generation in extractors)
- [x] Tableau
- [x] Compass (Resolves https://github.com/odpf/meteor/issues/474)
- [x] Also, adding struct tags(`json` and `yaml`) in provider configs required for marshalling into yaml

### CODE REVIEW

File: plugins/extractors/tableau/tableau.go [hunk 1]
```diff
@@ -22,7 +22,6 @@ var summary string
 var sampleConfig = `
 host: https://server.tableau.com
 version: "3.12"
-identifier: my-tableau
```
Comment (kushsharma) [inline plugins/extractors/tableau/tableau.go:25]: What is this used for?

File: plugins/sinks/compass/README.md [hunk 2]
```diff
@@ -13,8 +13,8 @@ sinks:
       compass-User-Email: meteor@odpf.io
       Header-1: value11,value12
     labels:
-      myCustom: $properties.attributes.myCustomField
-      sampleLabel: $properties.labels.sampleLabelField
+      myCustom: $attributes.myCustomField
```
Comment (kushsharma) [inline plugins/sinks/compass/README.md:16]: do these notations work?

File: test/e2e/mysql_kafka.yml [hunk 3]
```diff
@@ -4,7 +4,6 @@ source:
   name: mysql
   config:
     connection_url: root:admin@tcp(127.0.0.1:3306)/
-    identifier: test_db
```
Comment (kushsharma) [inline test/e2e/mysql_kafka.yml:7]: how was this used earlier?

**Comment by kushsharma** [comment]
Extra `yaml/json` struct tags look alright but I do see a few extra changes like tableu `identifier`, what are these for?

File: plugins/extractors/tableau/tableau.go (hunk 1 above)
Comment (Chief-Rishab) [reply plugins/extractors/tableau/tableau.go:25]: In this meteor PR, `identifier` field was added in extractors for URN generation
https://github.com/odpf/meteor/pull/354/files
Later there was an updation to generate URN in a different manner in this PR, deprecating the identifier field altogether.
https://github.com/odpf/meteor/pull/387/files

File: plugins/sinks/compass/README.md (hunk 2 above)
Comment (Chief-Rishab) [reply plugins/sinks/compass/README.md:16]: There was an issue regarding the same https://github.com/odpf/meteor/issues/474 and I saw the sample config to verify https://github.com/odpf/meteor/blob/main/plugins/sinks/compass/sink.go#L47.

File: test/e2e/mysql_kafka.yml (hunk 3 above)
Comment (Chief-Rishab) [reply test/e2e/mysql_kafka.yml:7]: same as mentioned above, was earlier used for URN generation

**Comment by kushsharma** [comment]
LGTM

**Review by kushsharma — APPROVED** [review]

**Pull request merged**